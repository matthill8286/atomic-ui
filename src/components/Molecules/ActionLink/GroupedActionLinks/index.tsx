import React, { FC } from 'react'
import { QuickAction } from '../ActionLink.interface'
import { getActionLinkType } from '../helper'
import { ActionLink } from '../'

export interface GroupedActionLinksProps {
  links: QuickAction[]
}

export const GroupedActionLinks: FC<GroupedActionLinksProps> = ({ links }) => {
  return (
    <>
      {links.map((quickAction, index) => {
        const linkProp = getActionLinkType(quickAction)
        return (
          <ActionLink
            key={index}
            target={quickAction.target}
            onClick={quickAction.onClick}
            {...linkProp}>
            {quickAction.actionLabel}
          </ActionLink>
        )
      })}
    </>
  )
}
