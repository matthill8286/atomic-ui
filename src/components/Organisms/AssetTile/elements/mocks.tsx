import { newAssetMocks, newStaticAssetMocks } from './mockAsset'
import { AssetTileLayout } from '@/components/Organisms/AssetTile/AssetTile.interface'

export const layout: AssetTileLayout[] = [
  AssetTileLayout.gridItem,
  AssetTileLayout.listItem,
  AssetTileLayout.compact,
  AssetTileLayout.auto,
]

export const mockAssets = {
  asset1: newAssetMocks[0],
  asset2: newAssetMocks[1],
  asset3: newAssetMocks[2],
  asset4: newAssetMocks[4], // screenshotOverride empty test, uses image as fallback
}

export const staticMockAssets = {
  asset1: newStaticAssetMocks[0],
  asset2: newStaticAssetMocks[1],
  asset3: newStaticAssetMocks[2],
  asset4: newStaticAssetMocks[3],
}

export const gridSelection = {
  '1 Item': 1,
  '2 Items': 2,
  '3 Items': 3,
}

export const assetNames = ['asset1', 'asset2', 'asset3']

export const assetOptions = {
  '(Automatic)': 'asset1',
  asset1: 'asset1',
  asset2: 'asset2',
  asset3: 'asset3',
}

export const featureOptions = {
  showSkills: 'showSkills',
  showBookmark: 'showBookmark',
  showLock: 'showLock',
  showProvider: 'showProvider',
  showAssetMeta: 'showAssetMeta',
  showFeatured: 'showFeatured',
}

export const baseProps = {}

export const presetProps = ['None', 'Reduced']
