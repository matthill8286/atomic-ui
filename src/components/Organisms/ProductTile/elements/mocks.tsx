import { newProductMocks, newStaticProductMocks } from './mockProduct'
import { ProductTileLayout } from '@/components/Organisms/ProductTile/ProductTile.interface'

export const layout: ProductTileLayout[] = [
  ProductTileLayout.gridItem,
  ProductTileLayout.listItem,
  ProductTileLayout.compact,
  ProductTileLayout.auto,
]

export const mockProducts = {
  product1: newProductMocks[0],
  product2: newProductMocks[1],
  product3: newProductMocks[2],
  product4: newProductMocks[4], // screenshotOverride empty test, uses image as fallback
}

export const staticMockProducts = {
  product1: newStaticProductMocks[0],
  product2: newStaticProductMocks[1],
  product3: newStaticProductMocks[2],
  product4: newStaticProductMocks[3],
}

export const gridSelection = {
  '1 Item': 1,
  '2 Items': 2,
  '3 Items': 3,
}

export const productNames = ['product1', 'product2', 'product3']

export const productOptions = {
  '(Automatic)': 'product1',
  product1: 'product1',
  product2: 'product2',
  product3: 'product3',
}

export const featureOptions = {
  showSkills: 'showSkills',
  showBookmark: 'showBookmark',
  showLock: 'showLock',
  showProvider: 'showProvider',
  showProductMeta: 'showProductMeta',
  showFeatured: 'showFeatured',
}

export const baseProps = {}

export const presetProps = ['None', 'Reduced']
