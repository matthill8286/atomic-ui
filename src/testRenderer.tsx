import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
