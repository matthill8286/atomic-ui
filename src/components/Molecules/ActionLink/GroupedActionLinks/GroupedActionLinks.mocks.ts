import { QuickAction } from '../ActionLink.interface'

export const actionLinks: QuickAction[] = [
  {
    actionLabel: 'Action Link 1',
    href: 'http://saiyan.content-learning.com',
    to: '',
    branded: true,
    target: '_blank',
    links: [],
  },
  {
    actionLabel: 'Action Link 2',
    href: 'http://saiyan.content-learning.com',
    to: '',
    branded: false,
    target: '_blank',
    links: [],
  },
]

export const actionToLinks: QuickAction[] = [
  {
    actionLabel: 'Action To Link 1',
    to: '/profile',
    href: null,
    branded: false,
    target: '_blank',
    links: [],
  },
  {
    actionLabel: 'Action To Link 2',
    to: '/history',
    href: '',
    branded: false,
    target: '_blank',
    links: [
      {
        actionLabel: 'Action Link 1',
        href: 'http://saiyan.content-learning.com',
        to: '',
        branded: true,
        target: '_blank',
        links: [],
      },
      {
        actionLabel: 'Action Link 2',
        href: 'http://saiyan.content-learning.com',
        to: '',
        branded: false,
        target: '_blank',
        links: [],
      },
    ],
  },
]
