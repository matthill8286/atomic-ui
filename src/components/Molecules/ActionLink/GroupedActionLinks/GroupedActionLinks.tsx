import React, { useEffect, useRef } from 'react'
import { breakpoints, css, styled } from '@/styles'
import { useDropdownState, useWindowDimensions } from '@/components/Helper'
import { StyledDropdown } from '@/components/Molecules/DropdownButton'
import { PaddingMap, ThemeColors, ThemeFontSizes, VerticalPaddingMap } from '@/types'
import { ActionLink } from '../ActionLink'
import { QuickAction } from '../ActionLink.interface'
import { getActionLinkType } from '../helper'

export type GroupedActionLinksProps = {
  links: QuickAction[]
  fontSize: ThemeFontSizes
  color: ThemeColors
  flexed?: boolean
  padding?: PaddingMap
  dropdownFlexed?: boolean
}

const StyledWrapper = styled.div<{
  flexed: boolean
  padding?: string | VerticalPaddingMap | undefined
}>(
  ({ theme, flexed }) => css`
    list-style: none;
    display: ${flexed ? `flex` : `inline-flex`};
    margin-top: 0;
    margin-right: ${!flexed ? theme.spacing.base.sm : 0};

    :hover {
      cursor: pointer;
      ${flexed && `background: ${theme.color.grey2};`};
    }
  `
)

const StyleDropdownWrapper = styled.div`
  display: flex;
`

const StyledReference = styled.div`
  position: relative;
  display: flex;
  flex-grow: 1;

  > a {
    flex-grow: 1;
  }
`

export const GroupedActionLinks: ({
  links,
  fontSize,
  color,
  padding,
  flexed,
  dropdownFlexed,
  testIdSuffix,
}: {
  links: QuickAction[]
  fontSize: ThemeFontSizes
  color: ThemeColors
  padding?: PaddingMap
  flexed?: boolean
  dropdownFlexed?: boolean
  testIdSuffix?: string
}) => JSX.Element = ({
  links,
  fontSize,
  color,
  padding,
  flexed = false,
  testIdSuffix,
  dropdownFlexed,
}) => {
  const { element, toggleDropdown, showDropdown } = useDropdownState<HTMLDivElement>()
  const { breakpoint: currentBreakpoint } = useWindowDimensions()
  const isMobile = currentBreakpoint < breakpoints.md
  const ref = useRef<HTMLElement>(null)

  const onToggleDropdown = () => {
    toggleDropdown()
  }

  const handleMenuItemClick = item => {
    if (showDropdown) {
      toggleDropdown()
    }
  }

  const handleClickOutside = (event: Event): void => {
    if (!ref.current?.contains(event.target as Node) && showDropdown) {
      toggleDropdown()
    }
  }

  useEffect(() => {
    document.addEventListener('click', handleClickOutside)

    return () => {
      document.removeEventListener('click', handleClickOutside)
    }
  })

  const ActionMenuContent = ({ navActionList }) => (
    <GroupedActionLinks
      flexed={dropdownFlexed}
      links={navActionList}
      fontSize="sm"
      color="grey5"
      key="GroupedActionLinks"
      padding={padding}
    />
  )

  return (
    <>
      {links.map((quickAction: QuickAction, index) => {
        const linkProp = getActionLinkType(quickAction)
        return (
          <StyledWrapper
            key={`Link-${index++}|${quickAction.id}`}
            padding={padding}
            flexed={flexed}>
            <StyledReference ref={element} data-test={`grouped-action-links-${testIdSuffix}`}>
              <ActionLink
                key={quickAction.id}
                inline={!flexed}
                color={color}
                fontSize={fontSize}
                branded={quickAction.branded}
                target={quickAction.target}
                onClick={quickAction?.links?.length ? onToggleDropdown : quickAction.onClick}
                {...(!quickAction?.links?.length ? linkProp : {})}>
                {quickAction.actionLabel}
              </ActionLink>
              {quickAction?.links?.length > 0 && (
                <StyleDropdownWrapper key={`InnerLink-${index++}|${quickAction.id}`}>
                  <StyledDropdown isMobile={isMobile} surface="white" showDropdown={showDropdown}>
                    <ActionMenuContent navActionList={quickAction.links} />
                  </StyledDropdown>
                </StyleDropdownWrapper>
              )}
            </StyledReference>
          </StyledWrapper>
        )
      })}
    </>
  )
}
