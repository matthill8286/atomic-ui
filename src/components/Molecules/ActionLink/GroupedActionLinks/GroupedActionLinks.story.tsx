import React from 'react'
import { Meta, Story } from '@storybook/react/types-6-0'
import { GroupedActionLinks, GroupedActionLinksProps } from './'
import { actionLinks } from './GroupedActionLinks.mocks'

// eslint-disable-next-line import/no-default-export
export default {
  title: 'Design System/Molecules/Action Link/Grouped',
  component: GroupedActionLinks,
  args: { links: actionLinks },
} as Meta

const Template: Story<GroupedActionLinksProps> = (args: GroupedActionLinksProps) => (
  <GroupedActionLinks {...args} />
)

export const Default = Template.bind({})
