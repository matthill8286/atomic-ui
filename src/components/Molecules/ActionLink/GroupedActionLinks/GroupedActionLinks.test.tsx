import React from 'react'
import { mountWithTheme, renderWithTheme } from '@/testRenderer'
import { GroupedActionLinks, GroupedActionLinksProps } from './'
import { ActionLink } from '..'
import { actionLinks } from './GroupedActionLinks.mocks'

describe('<GroupedActionLinks /> component', () => {
  const props: GroupedActionLinksProps = { links: actionLinks }
  const defaultGroupedActionLinks = <GroupedActionLinks {...props} />

  test('should render with 2 <ActionLink /> children', () => {
    const groupedActionLinks = renderWithTheme(defaultGroupedActionLinks)
    const wrapper = mountWithTheme(defaultGroupedActionLinks)
    expect(wrapper).toHaveLength(1)
    expect(wrapper.find(ActionLink)).toHaveLength(2)
    expect(groupedActionLinks).toMatchSnapshot()
  })
})
