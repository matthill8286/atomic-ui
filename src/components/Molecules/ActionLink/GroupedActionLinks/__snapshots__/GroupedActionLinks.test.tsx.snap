// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupedActionLinks /> component should render with 2 <ActionLink /> children 1`] = `
Array [
  .c3 {
  color: #999999;
  font-size: 16px;
  line-height: 24px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 svg {
  width: 25px;
  height: 25px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #00C8DC;
}

.c2 svg g,
.c2 svg p {
  fill: #00C8DC;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #999999;
  opacity: 1.0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  position: relative;
  padding: 16px;
}

.c0::after {
  content: '';
  position: absolute;
  display: block;
  bottom: -2px;
  height: 2px;
  left: 24px;
  width: calc( 100% - 24px );
  background-color: #999999;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
  -webkit-transition: -webkit-transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  -webkit-transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
}

.c0:hover::after {
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c1 {
  margin-left: 0px;
  margin-right: 4px;
}

.c1 svg {
  height: 20px;
  width: 20px;
}

.c4 {
  display: block;
  color: #999999;
}

<a
    className="c0"
    color="grey4"
    href="http://mediamarkt.de"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="c1"
      color="grey4"
    >
      <div
        className="c2"
        color="primary"
      >
        <svg
          height={32}
          viewBox="0 0 24 24"
          width={32}
        />
      </div>
    </span>
    <span
      className="c3 c4"
      color="grey4"
      fontSize="sm"
    >
      Action Link 1
    </span>
  </a>,
  .c3 {
  color: #999999;
  font-size: 16px;
  line-height: 24px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 svg {
  width: 25px;
  height: 25px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #00C8DC;
}

.c2 svg g,
.c2 svg p {
  fill: #00C8DC;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #999999;
  opacity: 1.0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  position: relative;
  padding: 16px;
}

.c0::after {
  content: '';
  position: absolute;
  display: block;
  bottom: -2px;
  height: 2px;
  left: 24px;
  width: calc( 100% - 24px );
  background-color: #999999;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
  -webkit-transition: -webkit-transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  -webkit-transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
}

.c0:hover::after {
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c1 {
  margin-left: 0px;
  margin-right: 4px;
}

.c1 svg {
  height: 20px;
  width: 20px;
}

.c4 {
  display: block;
  color: #999999;
}

<a
    className="c0"
    color="grey4"
    href="http://mediamarkt.de"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="c1"
      color="grey4"
    >
      <div
        className="c2"
        color="primary"
      >
        <svg
          height={32}
          viewBox="0 0 24 24"
          width={32}
        />
      </div>
    </span>
    <span
      className="c3 c4"
      color="grey4"
      fontSize="sm"
    >
      Action Link 2
    </span>
  </a>,
]
`;
